archive/install_shield_3_z.ksy

```diff
@@ -0,0 +1,484 @@
+meta:
+  id: install_shield_3_z
+  title: InstallShield 3 Z archive
+  application: InstallShield 3
+  # Not to be confused with the Unix compress or pack compression formats,
+  # which use the uppercase Z and lowercase z extensions respectively.
+  file-extension: z
+  xref:
+    justsolve: InstallShield_Z
+  license: MIT
+  ks-version: 0.9
+  imports:
+    - /common/dos_datetime_backwards
+  endian: le
+  bit-endian: le
+doc: |
+  Archive format used by InstallShield 3 installers.
+  File data can optionally be compressed using the PKWARE DCL Implode algorithm.
+  This is not to be confused with the old ZIP Implode compression algorithm of the same name.
+doc-ref:
+  - https://github.com/OpenRA/OpenRA/blob/fe146cb77a25091ed342e38104292ac3cb19019b/OpenRA.Mods.Common/FileSystem/InstallShieldPackage.cs
+  - https://github.com/OmniBlade/isextract/blob/5adb0af87fb0aaad0b436db0b7e7356947af3ee2/src/isextract.cpp
+  - https://github.com/wfr/unshieldv3/blob/0037ff581a4d862757ccc2c584aefdc0fe58a38d/installshieldarchivev3.cpp
+  - https://github.com/adrium/unshieldv3/blob/9f47d1c501abb778eb40beb7fdbb24ae556b7c46/installshieldarchivev3.cpp
+  - https://github.com/lephilousophe/idecomp/blob/354961c0cfe86632a39e4fea01e0ca521ffac3e0/idecomp.py
+  - https://github.com/putara/isdecomp
+  - https://github.com/agrif/unshield
+  - http://kannegieser.net/veit/quelle/stix_src.arj # STIX.PAS
+seq:
+  - id: magic
+    contents: [0x13, 0x5d, 0x65, 0x8c]
+  - id: len_header
+    type: u1
+    valid: sizeof<header>
```

Comment:
Why not to assumme that someone can reuse this format and extend it, so instead use `size` in `header` field?

Reply:
1. That is never going to happen for this format :)
2. Every file I tested with has the same value in the size field, so I'm not even sure if that is the field's actual meaning. [idecomp.py apparently thinks so](https://github.com/lephilousophe/idecomp/blob/354961c0cfe86632a39e4fea01e0ca521ffac3e0/idecomp.py#L266-L269), but I don't know where its author got that information from, or if they just guessed. So if there are any files where this field has a different value, I want the parsing to fail at first, because it's hard to say if the rest of the spec will behave properly in that case.
3. The idecomp.py code also only handles this exact header size, so this check probably won't cause any problems in practice.